sdk/core/azure-core/src/main/java/com/azure/core/util/logging/LoggingEventBuilder.java

```diff
@@ -0,0 +1,289 @@
+     *
+     * @param key String key.
+     * @param valueSupplier String value supplier function.
+     * @return The updated {@code LoggingEventBuilder} object.
+     */
+    public LoggingEventBuilder addKeyValue(String key, Supplier<String> valueSupplier) {
+        if (this.isEnabled) {
+            if (this.context == null) {
+                this.context = new ArrayList<>();
+            }
+
+            this.context.add(new ContextKeyValuePair(key, valueSupplier));
+        }
+        return this;
+    }
+
+    /**
+     * Logs message annotated with context.
+     *
+     * @param message the message to log.
+     */
+    public void log(String message) {
+        if (this.isEnabled) {
+            logger.performLogging(level, false, getMessageWithContext(message));
+        }
+    }
+
+    /**
+     * Logs message annotated with context.
+     *
+     * @param messageSupplier string message supplier.
+     */
+    public void log(Supplier<String> messageSupplier) {
+        if (this.isEnabled) {
+            String message = messageSupplier != null ? messageSupplier.get() : null;
+            logger.performLogging(level, false, getMessageWithContext(message));
+        }
+    }
+
+    /**
+     * Logs a format-able message that uses {@code {}} as the placeholder at {@code warning} log level.
+     *
+     * @param format The format-able message to log.
+     * @param args Arguments for the message. If an exception is being logged, the last argument should be the {@link
+     * Throwable}.
+     */
+    public void log(String format, Object... args) {
+        if (this.isEnabled) {
+            logger.performLogging(level, false, getMessageWithContext(format), args);
+        }
+    }
+
+    /**
+     * Logs the {@link Throwable} and returns it to be thrown.
+     *
+     * @param throwable Throwable to be logged and returned.
+     * @return The passed {@link Throwable}.
+     * @throws NullPointerException If {@code runtimeException} is {@code null}.
+     */
+    public Throwable log(Throwable throwable) {
+        Objects.requireNonNull(throwable, "'throwable' cannot be null.");
+
+        if (this.isEnabled) {
+            logger.performLogging(level, true, getMessageWithContext(throwable.getMessage()), throwable);
+        }
+
+        return throwable;
+    }
+
+    private String getMessageWithContext(String message) {
+        if (message == null) {
+            message = "";
+        }
+
+        int contextSize = context == null ? 0 : context.size();
+
+        StringBuilder sb = new StringBuilder(20 + contextSize * 20 + message.length());
+        sb.append("{\"")
+            // message must be first for log parsing tooling to work, key also works as a
+            // marker for Azure SDK logs so we'll write it even if there is no message
+            .append(AZURE_SDK_LOG_MESSAGE_KEY)
+            .append("\":\"")
+            .append(message)
+            .append("\",");
```

Comment:
Adding a `,` here may result in a trailing `,` if the contextSize is 0.  For e.g. 
```java
logger.atError().log(exception);
```
will result in the following json

```json
{
    "az.sdk.message": "exception message",
}
```